iter_markers_with_node() has double entries for async test functions

## Description

With `pytest-asyncio` and `async def test_...()` function decorated with `@pytest.mark.xyz` each marker appears twice in `request.node.iter_markers_with_node()`.

## Environment

`pip list`
```
Package        Version
-------------- -------
iniconfig      2.0.0
packaging      24.0
pip            23.3.1
pluggy         1.4.0
pytest         8.1.1
pytest-asyncio 0.23.6
```

`pytest.ini`
```
[pytest]
log_level = INFO
markers = 
    network: Requires live network
    slow: Slow to execute
    smoke: Smoke tests
asyncio_mode = auto
```

OS / Python
* cPython 3.12.2
* Linux / amd64

## Test

`test_iter_markers.py`
```python
import logging

import pytest

logger = logging.getLogger("markerdebug")


def check_markers(request: pytest.FixtureRequest) -> None:
    markers = []
    marker_ids = set()
    for node, marker in request.node.iter_markers_with_node():
        logging.info(
            "node_id=%r node=%r marker_id=%r marker=%r",
            id(node),
            node,
            id(marker),
            marker,
        )
        markers.append(marker)
        marker_ids.add(id(marker))
    assert len(markers) == len(marker_ids)


@pytest.mark.slow
@pytest.mark.network
@pytest.mark.usefixtures("monkeypatch")
async def test_debug_iter_markers_async(request: pytest.FixtureRequest) -> None:
   check_markers(request) 


@pytest.mark.slow
@pytest.mark.network
@pytest.mark.usefixtures("monkeypatch")
def test_debug_iter_markers_sync(request: pytest.FixtureRequest) -> None:
    check_markers(request)
```

## Output

```
============================================= test session starts =============================================
platform linux -- Python 3.12.2, pytest-8.1.1, pluggy-1.4.0
configfile: pytest.ini
plugins: asyncio-0.23.6
asyncio: mode=Mode.AUTO
collected 2 items                                                                                             

test_iter_markers.py F.                                                                                 [100%]

================================================== FAILURES ===================================================
________________________________________ test_debug_iter_markers_async ________________________________________

request = <FixtureRequest for <Coroutine test_debug_iter_markers_async>>

    @pytest.mark.slow
    @pytest.mark.network
    @pytest.mark.usefixtures("monkeypatch")
    async def test_debug_iter_markers_async(request: pytest.FixtureRequest) -> None:
>      check_markers(request)

test_iter_markers.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

...
>       assert len(markers) == len(marker_ids)
E       AssertionError: assert 7 == 4
E        +  where 7 = len([Mark(name='usefixtures', args=('monkeypatch',), kwargs={}), Mark(name='network', args=(), kwargs={}), Mark(name='slow...rgs=('monkeypatch',), kwargs={}), Mark(name='network', args=(), kwargs={}), Mark(name='slow', args=(), kwargs={}), ...])
E        +  and   4 = len({140114061819008, 140114061990688, 140114070935392, 140114075899504})

test_iter_markers.py:21: AssertionError
---------------------------------------------- Captured log call ----------------------------------------------
INFO     root:test_iter_markers.py:12 node_id=140114061847456 node=<Coroutine test_debug_iter_markers_async> marker_id=140114061990688 marker=Mark(name='usefixtures', args=('monkeypatch',), kwargs={})
INFO     root:test_iter_markers.py:12 node_id=140114061847456 node=<Coroutine test_debug_iter_markers_async> marker_id=140114070935392 marker=Mark(name='network', args=(), kwargs={})
INFO     root:test_iter_markers.py:12 node_id=140114061847456 node=<Coroutine test_debug_iter_markers_async> marker_id=140114061819008 marker=Mark(name='slow', args=(), kwargs={})
INFO     root:test_iter_markers.py:12 node_id=140114061847456 node=<Coroutine test_debug_iter_markers_async> marker_id=140114061990688 marker=Mark(name='usefixtures', args=('monkeypatch',), kwargs={})
INFO     root:test_iter_markers.py:12 node_id=140114061847456 node=<Coroutine test_debug_iter_markers_async> marker_id=140114070935392 marker=Mark(name='network', args=(), kwargs={})
INFO     root:test_iter_markers.py:12 node_id=140114061847456 node=<Coroutine test_debug_iter_markers_async> marker_id=140114061819008 marker=Mark(name='slow', args=(), kwargs={})
INFO     root:test_iter_markers.py:12 node_id=140114061847456 node=<Coroutine test_debug_iter_markers_async> marker_id=140114075899504 marker=Mark(name='asyncio', args=(), kwargs={})
=========================================== short test summary info ===========================================
FAILED test_iter_markers.py::test_debug_iter_markers_async - AssertionError: assert 7 == 4
========================================= 1 failed, 1 passed in 0.02s =========================================
```

```
(Pdb) pprint.pprint(markers)
[Mark(name='usefixtures', args=('monkeypatch',), kwargs={}),
 Mark(name='network', args=(), kwargs={}),
 Mark(name='slow', args=(), kwargs={}),
 Mark(name='usefixtures', args=('monkeypatch',), kwargs={}),
 Mark(name='network', args=(), kwargs={}),
 Mark(name='slow', args=(), kwargs={}),
 Mark(name='asyncio', args=(), kwargs={})]
```

--- a/docs/source/reference/changelog.rst
+++ b/docs/source/reference/changelog.rst
@@ -2,6 +2,15 @@
 Changelog
 =========
 
+0.23.8 (UNRELEASED)
+===================
+- Fixes a bug that caused duplicate markers in async tests `#813 <https://github.com/pytest-dev/pytest-asyncio/issues/813>`_
+
+Known issues
+------------
+As of v0.23, pytest-asyncio attaches an asyncio event loop to each item of the test suite (i.e. session, packages, modules, classes, functions) and allows tests to be run in those loops when marked accordingly. Pytest-asyncio currently assumes that async fixture scope is correlated with the new event loop scope. This prevents fixtures from being evaluated independently from the event loop scope and breaks some existing test suites (see `#706`_). For example, a test suite may require all fixtures and tests to run in the same event loop, but have async fixtures that are set up and torn down for each module. If you're affected by this issue, please continue using the v0.21 release, until it is resolved.
+
+
 0.23.7 (2024-05-19)
 ===================
 - Silence deprecation warnings about unclosed event loops that occurred with certain CPython patch releases `#817 <https://github.com/pytest-dev/pytest-asyncio/pull/817>`_

--- a/pytest_asyncio/plugin.py
+++ b/pytest_asyncio/plugin.py
@@ -405,7 +405,8 @@ class PytestAsyncioFunction(Function):
             keywords=function.keywords,
             originalname=function.originalname,
         )
-        subclass_instance.own_markers.extend(function.own_markers)
+        subclass_instance.own_markers = function.own_markers
+        assert subclass_instance.own_markers == function.own_markers
         subclassed_function_signature = inspect.signature(subclass_instance.obj)
         if "event_loop" in subclassed_function_signature.parameters:
             subclass_instance.warn(

--- a/tests/hypothesis/test_base.py
+++ b/tests/hypothesis/test_base.py
@@ -23,7 +23,7 @@ def test_hypothesis_given_decorator_before_asyncio_mark(pytester: Pytester):
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1)
 
 
@@ -67,7 +67,7 @@ def test_can_use_explicit_event_loop_fixture(pytester: Pytester):
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=2)
     result.stdout.fnmatch_lines(
         [

--- a/tests/markers/test_function_scope.py
+++ b/tests/markers/test_function_scope.py
@@ -197,3 +197,34 @@ def test_standalone_test_does_not_trigger_warning_about_no_current_event_loop_be
     )
     result = pytester.runpytest_subprocess("--asyncio-mode=strict")
     result.assert_outcomes(warnings=0, passed=1)
+
+
+def test_asyncio_mark_does_not_duplicate_other_marks_in_auto_mode(
+    pytester: Pytester,
+):
+    pytester.makeconftest(
+        dedent(
+            """\
+            def pytest_configure(config):
+                config.addinivalue_line(
+                    "markers", "dummy_marker: mark used for testing purposes"
+                )
+            """
+        )
+    )
+    pytester.makepyfile(
+        dedent(
+            """\
+            import pytest
+
+            @pytest.mark.dummy_marker
+            async def test_markers_not_duplicated(request):
+                markers = []
+                for node, marker in request.node.iter_markers_with_node():
+                    markers.append(marker)
+                assert len(markers) == 2
+            """
+        )
+    )
+    result = pytester.runpytest_subprocess("--asyncio-mode=auto")
+    result.assert_outcomes(warnings=0, passed=1)

--- a/tests/markers/test_function_scope.py
+++ b/tests/markers/test_function_scope.py
@@ -43,7 +43,7 @@ def test_function_scope_supports_explicit_event_loop_fixture_request(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         '*is asynchronous and explicitly requests the "event_loop" fixture*'

--- a/tests/markers/test_module_scope.py
+++ b/tests/markers/test_module_scope.py
@@ -48,7 +48,7 @@ def test_asyncio_mark_works_on_module_level(pytester: Pytester):
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=2, warnings=2)
     result.stdout.fnmatch_lines(
         '*is asynchronous and explicitly requests the "event_loop" fixture*'

--- a/tests/modes/test_strict_mode.py
+++ b/tests/modes/test_strict_mode.py
@@ -79,7 +79,7 @@ def test_strict_mode_ignores_unmarked_coroutine(testdir):
         """
         )
     )
-    result = testdir.runpytest("--asyncio-mode=strict", "-W default")
+    result = testdir.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(skipped=1, warnings=1)
     result.stdout.fnmatch_lines(["*async def functions are not natively supported*"])
 
@@ -100,7 +100,7 @@ def test_strict_mode_ignores_unmarked_fixture(testdir):
         """
         )
     )
-    result = testdir.runpytest("--asyncio-mode=strict", "-W default")
+    result = testdir.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(skipped=1, warnings=2)
     result.stdout.fnmatch_lines(
         [

--- a/tests/test_asyncio_mark.py
+++ b/tests/test_asyncio_mark.py
@@ -15,7 +15,7 @@ def test_asyncio_mark_on_sync_function_emits_warning(pytester: Pytester):
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1)
     result.stdout.fnmatch_lines(
         ["*is marked with '@pytest.mark.asyncio' but it is not an async function.*"]
@@ -36,7 +36,7 @@ def test_asyncio_mark_on_async_generator_function_emits_warning_in_strict_mode(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(xfailed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ["*Tests based on asynchronous generators are not supported*"]
@@ -54,7 +54,7 @@ def test_asyncio_mark_on_async_generator_function_emits_warning_in_auto_mode(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=auto", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=auto", "-W default")
     result.assert_outcomes(xfailed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ["*Tests based on asynchronous generators are not supported*"]
@@ -76,7 +76,7 @@ def test_asyncio_mark_on_async_generator_method_emits_warning_in_strict_mode(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(xfailed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ["*Tests based on asynchronous generators are not supported*"]
@@ -96,7 +96,7 @@ def test_asyncio_mark_on_async_generator_method_emits_warning_in_auto_mode(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=auto", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=auto", "-W default")
     result.assert_outcomes(xfailed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ["*Tests based on asynchronous generators are not supported*"]
@@ -119,7 +119,7 @@ def test_asyncio_mark_on_async_generator_staticmethod_emits_warning_in_strict_mo
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(xfailed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ["*Tests based on asynchronous generators are not supported*"]
@@ -139,7 +139,7 @@ def test_asyncio_mark_on_async_generator_staticmethod_emits_warning_in_auto_mode
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=auto", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=auto", "-W default")
     result.assert_outcomes(xfailed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ["*Tests based on asynchronous generators are not supported*"]

--- a/tests/test_event_loop_fixture_finalizer.py
+++ b/tests/test_event_loop_fixture_finalizer.py
@@ -84,7 +84,7 @@ def test_event_loop_fixture_finalizer_handles_loop_set_to_none_async_with_fixtur
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         '*is asynchronous and explicitly requests the "event_loop" fixture*'
@@ -113,7 +113,7 @@ def test_event_loop_fixture_finalizer_raises_warning_when_fixture_leaves_loop_un
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W", "default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W", "default")
     result.assert_outcomes(passed=1, warnings=2)
     result.stdout.fnmatch_lines("*unclosed event loop*")
 
@@ -135,6 +135,6 @@ def test_event_loop_fixture_finalizer_raises_warning_when_test_leaves_loop_unclo
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W", "default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W", "default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines("*unclosed event loop*")

--- a/tests/test_event_loop_fixture_override_deprecation.py
+++ b/tests/test_event_loop_fixture_override_deprecation.py
@@ -22,7 +22,7 @@ def test_emit_warning_when_event_loop_fixture_is_redefined(pytester: Pytester):
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ["*event_loop fixture provided by pytest-asyncio has been redefined*"]
@@ -50,7 +50,7 @@ def test_emit_warning_when_event_loop_fixture_is_redefined_explicit_request(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=2)
     result.stdout.fnmatch_lines(
         ["*event_loop fixture provided by pytest-asyncio has been redefined*"]
@@ -80,7 +80,7 @@ def test_does_not_emit_warning_when_no_test_uses_the_event_loop_fixture(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict")
     result.assert_outcomes(passed=1, warnings=0)
 
 
@@ -107,5 +107,5 @@ def test_emit_warning_when_redefined_event_loop_is_used_by_fixture(pytester: Pyt
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)

--- a/tests/test_explicit_event_loop_fixture_request.py
+++ b/tests/test_explicit_event_loop_fixture_request.py
@@ -17,7 +17,7 @@ def test_emit_warning_when_event_loop_is_explicitly_requested_in_coroutine(
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ['*is asynchronous and explicitly requests the "event_loop" fixture*']
@@ -39,7 +39,7 @@ def test_emit_warning_when_event_loop_is_explicitly_requested_in_coroutine_metho
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ['*is asynchronous and explicitly requests the "event_loop" fixture*']
@@ -62,7 +62,7 @@ def test_emit_warning_when_event_loop_is_explicitly_requested_in_coroutine_stati
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ['*is asynchronous and explicitly requests the "event_loop" fixture*']
@@ -88,7 +88,7 @@ def test_emit_warning_when_event_loop_is_explicitly_requested_in_coroutine_fixtu
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ['*is asynchronous and explicitly requests the "event_loop" fixture*']
@@ -114,7 +114,7 @@ def test_emit_warning_when_event_loop_is_explicitly_requested_in_async_gen_fixtu
             """
         )
     )
-    result = pytester.runpytest("--asyncio-mode=strict", "-W default")
+    result = pytester.runpytest_subprocess("--asyncio-mode=strict", "-W default")
     result.assert_outcomes(passed=1, warnings=1)
     result.stdout.fnmatch_lines(
         ['*is asynchronous and explicitly requests the "event_loop" fixture*']

